docs/repo-specific-configuration.md

```diff
@@ -52,6 +52,12 @@ updates.pin  = [ { groupId = "com.example", artifactId="foo", version = "1.1." }
 # Defaults to empty `[]` which mean Scala Steward will not ignore dependencies.
 updates.ignore = [ { groupId = "org.acme", artifactId="foo", version = "1.0" } ]
 
+# The dependencies which match the given patterns are allowed to be updated to pre-release from stable.
+#
+# Each pattern must have `groupId`, and may have `artifactId` and `version`.
+# Defaults to empty `[]` which mean Scala Steward will ignore all stable to pre-release update options.
+updates.allowPreRelease  = [ { groupId = "com.example", artifactId="foo" } ]
```

Comment:
Question: if I add this, do I get Ms and RCs or do I get snapshots? What if only want Ms and RCs?

Reply:
For now both. Could you help me understand your requirements a bit better. Instead of boolean flag we would like to have pre-release modes like:

1. none - previous behavior
2. only Ms and RCs
3. only snapshots
4. all kind of pre-releases (2 + 3)

Is this correct? Or should it be 

1. none - previous behavior
2. only Ms and RCs
3. Ms, RCs and snapshots

It will end up having 2 lists in seconds anyway:
* updates.allowPreRelease
* updates.allowSnapshots

The question is "Should allowSnapshots include allowPreRelease?"

Reply:
Yes, I think `allowSnapshots` should imply `allowPreRelease`. Similarly, it should allow updates to ordinary stable releases! :) For example, if a pre-release or snapshot is succeeded by a stable release, it should update to that (at least until the next snapshot/pre-release is available).

Reply:
@arixmkii 
Could you add a test case for below?

> if a pre-release or snapshot is succeeded by a stable release, it should update to that

Reply:
I agree with what @armanbilge said in https://github.com/scala-steward-org/scala-steward/pull/2635#discussion_r900295436. But I would prefer implementing `allowSnapshots` in a later PR so that this PR does not become too big. I think there is no harm if `allowPreRelease` implies `allowSnapshots` for now.

Btw, if we pluralize snapshots in `allowSnapshots` shouldn't we also pluralize pre-releases in `allowPreReleases`?

Reply:
Implemented requested changes:
* used plural version of preReleases, where appropriate
* added more tests and grouped them logically
* added filter to prohibit updates from regular pre-release to snapshot

Rebased to latest main and squashed commits.